airbyte-config/config-models/src/main/java/io/airbyte/config/helpers/StateMessageHelper.java

```diff
@@ -39,14 +65,10 @@ public static Optional<StateWrapper> getTypedState(final JsonNode state) {
         return Optional.of(getLegacyStateWrapper(state));
```

Comment:
I know this isn't part of your PR's changes, but I read over this line again and I'm not sure if it is correct. This if statement seems to be saying that if any state message in the deserialized list has any additional properties set, we just immediately return a legacy state object. 

Why? I'm not seeing why the presence of additional properties means we should default to returning legacy here.

Reply:
Updated after our discussion from this morning